src/backends/plonky2/primitives/ec/curve.rs

```diff
@@ -121,6 +122,27 @@ impl fmt::Display for Point {
     }
 }
 
+impl FromStr for Point {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let point_bytes = bs58::decode(s)
```

Comment:
This code is now duplicated with `Deserialize for Point`.  Could you update the `Deserialize` to be something like this?
```rust
let point_b58 = String::deserialize(deserializer)?;
Ok(Point::from_str(point_b58)?)
```